config/routes.rb

```diff
@@ -3,6 +3,9 @@
   namespace :hq do
     resources :partners, except: [:destroy] do
       resources :orphan_lists, only: [:index]
+      resources :pending_orphan_lists, only: [:create, :new, :destroy] do
+        ['upload', 'validate', 'import'].each { |action| get action, on: :collection }
```

Comment:
the get is hidden in the logic, i know its more verbose but I find it easier to understand
```
collection do
  get :upload
  get :validate
  get :import
end
```
additionally once uploaded are the `validate` and `import` actions member actions ?